src/NumericalAlgorithms/Interpolation/PredictedZeroCrossing.cpp

```diff
@@ -6,33 +6,56 @@
 #include <deque>
 #include <vector>
 
-#include "NumericalAlgorithms/Interpolation/LinearLeastSquares.hpp"
+#include "NumericalAlgorithms/Interpolation/LinearRegression.hpp"
 
 namespace intrp {
 
 double predicted_zero_crossing_value(const std::vector<double>& x_values,
                                      const std::vector<double>& y_values) {
-  intrp::LinearLeastSquares<1> predictor{x_values.size()};
-  const auto coefficients = predictor.fit_coefficients(x_values, y_values);
-  return -coefficients[0]/coefficients[1];
+  const auto [intercept, slope, delta_intercept, delta_slope] =
+      linear_regression(x_values, y_values);
+
+  // See the doxygen comments for details of the implementation.
+  // For now, set all the x's to zero if denominators are zero;
+  // the actual return type in this case will be determined below.
```

Comment:
I don't understand this comment.  I assume "return type" is supposed to be something else.

Reply:
I removed the last part of the comment (which didn't make sense) and squashed.